src/main/java/org/jabref/logic/importer/fetcher/ISIDOREFetcher.java

```diff
@@ -0,0 +1,249 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import org.jabref.logic.help.HelpFile;
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.logic.importer.IdBasedParserFetcher;
+import org.jabref.logic.importer.Parser;
+import org.jabref.logic.net.URLDownload;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.types.EntryType;
+import org.jabref.model.entry.types.StandardEntryType;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * Fetcher for ISIDORE
+ * Will take in the link to the website or the last six digits that identify the reference
+ * Uses ISIDORE's API.
+ */
+public class ISIDOREFetcher implements IdBasedParserFetcher {
+    private String URL;
+
+    public ISIDOREFetcher() {
+    }
+
+    @Override
+    public URL getUrlForIdentifier(String identifier) throws URISyntaxException, MalformedURLException, FetcherException {
+        identifier = identifier.trim();
+        // this allows the user to input only the six-digit code at the end.
+        if (identifier.length() == 6) {
+            identifier = "https://isidore.science/document/10670/1." + identifier;
+        }
+        // allows the user to put in the eight digits including the "1."
+        if (identifier.length() == 8) {
```

Comment:
I would merge the statements

```suggestion
        } else if (identifier.length() == 8) {
            // allows the user to put in the eight digits including the "1."
```